src/main/kotlin/teaforge/platform/RoboRio/Main.kt

```diff
@@ -90,17 +140,17 @@ enum class PwmPort {
     Nine,
 }
 
-enum class DioPort {
-    Zero,
-    One,
-    Two,
-    Three,
-    Four,
-    Five,
-    Six,
-    Seven,
-    Eight,
-    Nine,
+enum class DioPort(val id: Int) {
+    Zero(0),
+    One(1),
+    Two(2),
+    Three(3),
+    Four(4),
+    Five(5),
+    Six(6),
+    Seven(7),
+    Eight(8),
+    Nine(9),
```

Comment:
I assume this is done so that function is not needed elsewhere to convert to an Int?

I understand the appeal but it has a downside, it's now possible to construct `Three(5)`. It also ties the logical value `Three` to a specific representation `3` but there may be multiple representations (a string that is displayed, a JSON or ProtoBuf encoding for transfer over a network, etc.). If there's just the logical value `Three` and then various functions that are `DioPort` => whatever none of them are easier or harder to add or modify. This isn't a problem at the beginning when we are 100% sure we know exactly how this will be used, but it's a nice thing to have later when we find exceptions to things we thought would never change.